tests/benchmark/bm_vecsim_basics.h

```diff
@@ -94,19 +152,88 @@ void BM_VecSimBasics<index_type_t>::DeleteLabel(algo_t *index, benchmark::State
     }
     memory_delta = index->getAllocationSize() - memory_before;
 
+    BM_VecSimGeneral::thread_pool_wait();
+    // Remove the rest of the vectors that hadn't been swapped yet for tiered index.
+    if (VecSimIndex_BasicInfo(index).algo == VecSimAlgo_TIERED) {
+        reinterpret_cast<TieredHNSWIndex<data_t, data_t> *>(index)->pendingSwapJobsThreshold = 1;
+        reinterpret_cast<TieredHNSWIndex<data_t, data_t> *>(index)->executeReadySwapJobs();
```

Comment:
separate `executeReadySwapJobs` into 2 functions:1. `executeReadySwapJobsIMP` that executes, 
2. `executeReadySwapJobs` -checks the condition and calls `executeReadySwapJobsIMP`
and here call executeReadySwapJobsIMP instead of changing `pendingSwapJobsThreshold`

Reply:
I just moved the check out from `executeReadySwapJobs`, now we can run it directly